compiler-rt/lib/profile/InstrProfilingUtil.c

```diff
@@ -373,3 +373,72 @@ COMPILER_RT_VISIBILITY int lprofReleaseMemoryPagesToOS(uintptr_t Begin,
   return 0;
 #endif
 }
+
+#ifdef _AIX
+typedef struct fn_node {
+  AtExit_Fn_ptr func;
+  struct fn_node *next;
+} fn_node;
+typedef struct {
+  fn_node *top;
+} fn_stack;
+
+static void fn_stack_push(fn_stack *, AtExit_Fn_ptr);
+static AtExit_Fn_ptr fn_stack_pop(fn_stack *);
+/* return 1 if stack is empty, 0 otherwise */
+static int fn_stack_is_empty(fn_stack *);
+
+static fn_stack AtExit_stack = {0};
```

Comment:
why not just `static fn_node* AtExit_stack = NULL;`

Reply:
hmm, if I go with your suggestion I would have to either (a) init the stack before first use (I don't think there's a guaranteed moment in time where that's possible), or (b) guard every access of the pointer with an init-if-null sequence.
I'm open to suggestions.